lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -278,32 +294,42 @@ def _on_update_status(self, event):
 class OpenSearchPluginError(OpenSearchError):
     """Exception thrown when an opensearch plugin is invalid."""
 
-
-class OpenSearchPluginRelationClusterNotReadyError(OpenSearchPluginError):
-    """Exception thrown when making API calls and cluster is not yet ready.
-
-    This exception in specific should be handled by classes managing the relations of plugins.
-    """
+    def __init__(self, msg):
+        super().__init__(msg)
 
 
 class OpenSearchPluginMissingDepsError(OpenSearchPluginError):
     """Exception thrown when an opensearch plugin misses installed dependencies."""
 
+    def __init__(self, name, deps):
+        super().__init__(f"Failed to install plugin: {name} - missing dependencies {deps}")
+        self.deps = deps
+
 
 class OpenSearchPluginInstallError(OpenSearchPluginError):
     """Exception thrown when opensearch plugin installation fails."""
 
+    def __init__(self, name):
+        super().__init__("Failed to install plugin: {}".format(name))
+
 
 class OpenSearchPluginRemoveError(OpenSearchPluginError):
     """Exception thrown when opensearch plugin removal fails."""
 
+    def __init__(self, name):
+        super().__init__("Failed to remove plugin: {}".format(name))
+
 
 class OpenSearchPluginMissingConfigError(OpenSearchPluginError):
     """Exception thrown when config() or disable() fails to find a config key.
 
     The plugin itself should raise a KeyError, to avoid burden in the plugin development.
     """
 
+    def __init__(self, name, configs: List[str]):
+        super().__init__(f"Plugin {name} is missing configs: {configs}")
+        self.configs = configs
```

Comment:
same as above, safe to delete unless needed for something?

Reply:
It doesn't seem like we need this constructor? we don't need to keep these attributes as they're only used for logging. Let's remove this constructor altogether and just set the right message with the list of configs in the str.